src/module.c

```diff
@@ -7874,6 +7877,42 @@ int RM_ModuleTypeReplaceValue(RedisModuleKey *key, moduleType *mt, void *new_val
     return REDISMODULE_OK;
 }
 
+/* For a specified command, parse its arguments and return an array the
+ * contains the indexes of all keys. This function is essnetially a
+ * more efficient way to do COMMAND GETKEYS.
+ *
+ * A NULL value indicates no keys, or an unknown command.
+ *
+ * NOTE: The returned array is not a Redis Module object so it does not
+ * get automatically freed even when auto-memory is used. The caller
+ * must explicitly call RM_Free() to free it.
+ */
+int *RM_GetCommandKeys(RedisModuleCtx *ctx, const char *cmdname, RedisModuleString **argv, int argc, int *num_keys) {
+    UNUSED(ctx);
+    struct redisCommand *cmd;
+    int *res = NULL;
+
+    if ((cmd = lookupCommandByCString(cmdname)) == NULL)
+        return NULL;
+
+    getKeysResult result = GETKEYS_RESULT_INIT;
+    getKeysFromCommand(cmd, argv, argc, &result);
```

Comment:
I think we must do an arity check before calling it.
some of the implementations of the command key callbacks don't do that check. AFAIK all the other places that use it are on a `cmd` that already passed this check.